### PR TITLE
Fix type-declarations

### LINKED ISSUE #21: TypeScript Declaration is broken
The current type-declaration require `esModuleInterop` top be enabled (which is disabled by default - the decision whether or not to use `esmoduleInterop` should be left to the user consuming this module).

### PR DESCRIPTION
**Pull request by manuth**
This PR fixes #21 by applying some tiny modifications to the module-imports and adding the missing `@types/node`-dependency.

### CODE REVIEW

**Comment by manuth** [comment]
I'll take some minutes to add a test-case for this.
Wish me luck ✨

File: package.json [hunk 1]
```diff
@@ -16,6 +16,7 @@
     "dist/index.d.ts"
   ],
   "dependencies": {
+    "@types/node": "^15.6.1",
```
Comment (mshima) [inline package.json:19]: Needs to be a prod dependency?

Comment (manuth) [reply package.json:19]: Yeah - this needs to be installed for `mem-fs`es type-declaration to be intact.

That confused me at first, too, tbh 😂

Comment (manuth) [reply package.json:19]: Oh - I just noticed, that `@types/vinyl` is being used in the type-declaration, too, and is missing from the `dependencies`-array - I'll fix it in this PR as well

**Comment by manuth** [comment]
I just noticed that, with `esModuleInterop` enabled, the packages `@types/vinyl` and `@types/vinyl-file` aren't working and cause errors when running `tsc`.

I assume that's why type-checking for the node-modules are disabled as seen in the `tsconfig.json`:
https://github.com/SBoudrias/mem-fs/blob/f09f524b2c306300fef6f3b112d329fb541bda3f/tsconfig.json#L7

Though, I have absolutely no plan why said packages work without `esModuleInterop` enabled but, then again, do not work with `esModuleInterop` enabled, I'll just go on and disable it for `mem-fs` in order to make it work.

**Comment by manuth** [comment]
Alright! I was able to create a test now which proofs, that the module did not work without `esModuleInterop` enabled before.

Also, the fact, that `skipLibCheck` is required if `esModuleInterop` is enabled, was caused by `@types/vinyl-file`:
https://github.com/DefinitelyTyped/DefinitelyTyped/issues/53389

FYI - I created a PR to fix this issue in `@types/vinyl-file`
https://github.com/DefinitelyTyped/DefinitelyTyped/pull/53390

File: test/index.js [hunk 2]
```diff
@@ -143,4 +145,22 @@ describe('mem-fs', function () {
       });
     });
   });
+
+  describe('type-declaration', function() {
```
Comment (SBoudrias) [inline test/index.js:149]: Let's not run this through mocha. Instead, how about we just call `tsc -p ./folder/...` from the `npm test` command.

File: test/fixtures/tsconfig.default.json [hunk 3]
```diff
@@ -0,0 +1,9 @@
+{
```
Comment (SBoudrias) [inline test/fixtures/tsconfig.default.json:1]: Let's move those files in some type test specific folder. Maybe just `<root>/type-test`

**Comment by manuth** [comment]
Will do

File: test/index.js [hunk 4]
```diff
@@ -1,6 +1,8 @@
 'use strict';
 
 var assert = require('assert');
+var child_process = require('child_process');
+var npmWhich = require('npm-which');
```
Comment (mshima) [inline test/index.js:5]:
```suggestion
```

File: package.json [hunk 5]
```diff
@@ -16,13 +16,15 @@
...
+    "@types/vinyl": "^2.0.4",
     "vinyl": "^2.0.1",
     "vinyl-file": "^3.0.0"
   },
   "devDependencies": {
-    "@types/vinyl": "^2.0.4",
     "@types/vinyl-file": "^3.0.0",
     "mocha": "^8.3.2",
+    "npm-which": "^3.0.1",
```
Comment (mshima) [inline package.json:27]:
```suggestion
```

**Review by mshima — COMMENTED** [review]
There are some leftovers

**Comment by manuth** [comment]
Thanks for looking through it - I'll remove the leftovers asap!

**Review by SBoudrias — APPROVED** [review]

**Pull request merged**